src/sentry/utils/snowflake.py

```diff
@@ -55,3 +67,28 @@ def snowflake_id_generation():
     ID_VALIDATOR.validate(snowflake_id)
 
     return snowflake_id
+
+
+def get_redis_cluster(redis_key: str):
+    return redis.clusters.get("default").get_local_client_for_key(redis_key)
+
+
+def get_sequence_value_from_redis(redis_key: str, starting_timestamp: int) -> Tuple[int, int]:
+    cluster = get_redis_cluster(redis_key)
+
+    for i in range(int(_TTL.total_seconds())):
+        timestamp = starting_timestamp - i
+
+        sequence_value = cluster.get(str(timestamp))
+
+        if not sequence_value:
+            sequence_value = 0
+            cluster.setex(timestamp, int(_TTL.total_seconds()), sequence_value)
+            return timestamp, sequence_value
```

Comment:
`incr()` should initialize the value to 0 and then increment yielding `1`. You could do something like

```python
value = cluster.incr(timestamp)
if value == 1:
   cluster.expire(timestamp, int(_TTL.total_seconds()))
   return timestamp, value
...
```
This would saves you a redis operation.

Reply:
I thought of doing that as well but this will start our region sequence at 1 and not 0. would it be better to have 1 more region per second with the extra Redis operation or have 15 instead of 16 sequence values. Alternatively, is it janky to do something like this which will save us a Redis operation and still have 16 sequence values?
`
value = cluster.incr(timestamp)
value -= 1
if value == 0:
   cluster.expire(timestamp, int(_TTL.total_seconds()))
return timestamp, value
`